architecture/platforms/versal_acap.rst

```diff
@@ -0,0 +1,125 @@
+
+The services offered by PLM firmware are decided at build time: it is therefore important that the PLM firmware build configuration enables the services that OP-TEE will require.
+
+As an example, if OP-TEE is configured to generate a hardware unique key, it  will need access to the PLM Physical Unclonable Function and NVM services.
+
+.. note::
+   Communication between OP-TEE and the PLM uses the IPI mailbox controller being the IPI used selectable via ``CFG_VERSAL_MBOX_IPI_ID``.
+
+As described in freely available documentation for the Versal ACAP `boot-flow`_, the BootROM handles loading the PLM, while the PLM will handle loading the rest of the images including OP-TEE.
+
+A reference BIF file supporting an OP-TEE instance capable of loading the FPGA pdi can be seen below. In this example OP-TEE should be configured with ``CFG_DT=y`` and ``CFG_DT_ADDR=0x00001000``.
+If enabled the platform expects the FPGA bitstream at 0x40000000; the location is configurable using ``CFG_VERSAL_FPGA_DDR_ADDR``.
+
+.. code-block:: none
+
+	ROM_image:
+	{
+		image {
+                      { type=bootimage, file=vpl_gen_fixed.pdi }
+	              { type=bootloader, file=plm.elf }
+	              { core=psm, file=psmfw.elf }
+	        }
+	        image {
+	              id = 0x1c000000, name=apu_subsystem
+	              { type=raw, load=0x00001000, file=versal-vck190-revA-x-ebm-01-revA.dtb }
+	              { type=raw, load=0x40000000, file=fpga.pdi }
+	              { core=a72-0, exception_level=el-3, trustzone, file=bl31.elf }
+ 	              { core=a72-0, exception_level=el-2, file=u-boot.elf }
+	              { core=a72-0, exception_level=el-1, trustzone, load=0x60000000, startup=0x60000000, file=tee-raw.bin }
+	         }
+         }
+
+
+To build the boot-able image AMD/Xilinx uses the `bootgen tool`_; this tool aggregates all the different images in a single binary.
+
+The available configuration options depend on the architecture.
+
+In the case of Versal ACAP using the previously mentioned BIF file, the boot.bin generation would be as follows:
+
+.. code-block:: none
+		
+        $ bootgen -arch versal -image file.bif -o boot.bin
+   
+
+Cryptographic driver
+********************
+
+The Versal ACAP cryptography driver rests on the PLM's `xilsecure`_ service.
+It provides hardware assisted support for:
+
+    1. SHA3-384
+    2. RSA 2048, 4096   
+    3. ECC sign/verify
+    4. AES-GCM   
+
+Other drivers
+*************
+
+The ``Versal ACAP eFUSE`` driver uses the PLM `xilnvm`_ service.
+Access to certain eFuses require specific PLM configuration flags not selectable at run-time.
+
+.. note::
+   It is therefore left down to the user to make sure that the PLM has been configured as expected. 
+
+The ``Versal ACAP PUF`` driver uses the PLM `xilpuf`_ service.
+
+At the time of this writing, the platform support includes three native drivers:
+
+    1. ``Mailbox driver``
```

Comment:
Need more details for 1 2 and 3.

Reply:
why? to what end?

Reply:
Why  - is so customers know what they are used for. e.g. mailbox for what? Talking to FPGA? Other processors? etc. In this case it's specifically for OP-TEE to PLM communications. Just need a few sentences to describe the drivers.  Also, no mention of how you linked the TRNG  into the Linux. That's a really cool feature that's not documented but very useful.